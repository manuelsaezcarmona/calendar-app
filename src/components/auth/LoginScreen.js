/* eslint-disable object-curly-newline */
import React from 'react';
// eslint-disable-next-line import/no-unresolved
import './login.css';
import { useDispatch } from 'react-redux';
import Swal from 'sweetalert2';
import { useForm } from '../../hooks/useForm';
import {
  StartLogin,
  startRegister
} from '../../redux/actioncreators/auth.actioncreator';

export function LoginScreen() {
  const dispatch = useDispatch();
  const [formLoginValues, handleLoginInputChange] = useForm({
    logEmail: '',
    logPassw: ''
  });
  const { logEmail, logPassw } = formLoginValues;

  const [formRegisterValues, handleRegisterInputChange] = useForm({
    RegName: '',
    regEmail: '',
    regPassw1: '',
    regPassw2: ''
  });

  const { RegName, regEmail, regPassw1, regPassw2 } = formRegisterValues;

  const handleLoginSubmit = (e) => {
    e.preventDefault();

    dispatch(StartLogin(logEmail, logPassw));
  };

  // eslint-disable-next-line consistent-return
  const handleRegisterSubmit = (e) => {
    e.preventDefault();

    if (regPassw1 !== regPassw2) {
      return Swal.fire('Error', 'las contrase単as deben ser iguales', 'error');
    }

    dispatch(startRegister(regEmail, regPassw1, RegName));
  };

  return (
    <div className="container login-container">
      <div className="row">
        <div className="col-md-6 login-form-1">
          <h3>Ingreso</h3>
          <form onSubmit={handleLoginSubmit}>
            <div className="form-group">
              <input
                type="text"
                className="form-control"
                placeholder="Correo"
                name="logEmail"
                value={logEmail}
                onChange={handleLoginInputChange}
              />
            </div>
            <div className="form-group">
              <input
                type="password"
                className="form-control"
                placeholder="Contrase単a"
                name="logPassw"
                value={logPassw}
                onChange={handleLoginInputChange}
              />
            </div>
            <div className="form-group">
              <input type="submit" className="btnSubmit" value="Login" />
            </div>
          </form>
        </div>

        <div className="col-md-6 login-form-2">
          <h3>Registro</h3>
          <form onSubmit={handleRegisterSubmit}>
            <div className="form-group">
              <input
                type="text"
                className="form-control"
                placeholder="Nombre"
                name="RegName"
                value={RegName}
                onChange={handleRegisterInputChange}
              />
            </div>
            <div className="form-group">
              <input
                type="email"
                className="form-control"
                placeholder="Correo"
                name="RegEmail"
                value={regEmail}
                onChange={handleRegisterInputChange}
              />
            </div>
            <div className="form-group">
              <input
                type="password"
                className="form-control"
                placeholder="Contrase単a"
                name="regPassw1"
                value={regPassw1}
                onChange={handleRegisterInputChange}
              />
            </div>

            <div className="form-group">
              <input
                type="password"
                className="form-control"
                placeholder="Repita la contrase単a"
                name="regPassw2"
                value={regPassw2}
                onChange={handleRegisterInputChange}
              />
            </div>

            <div className="form-group">
              <input type="submit" className="btnSubmit" value="Crear cuenta" />
            </div>
          </form>
        </div>
      </div>
    </div>
  );
}
