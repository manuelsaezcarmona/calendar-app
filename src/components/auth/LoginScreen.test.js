import React from 'react';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
// esto es para tener la ayuda de los metodos de jest
import '@testing-library/jest-dom';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import { LoginScreen } from './LoginScreen';

/* jest.mock('../../redux/actioncreators/event.actioncreator', () => ({
  startEventDelete: jest.fn()
})); */

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

const initState = {};

const store = mockStore(initState);
// En el componente disparo una accion (asincrona) que ya testee en las actions.
// voy a hacer un mock de esa accion.
store.dispatch = jest.fn();

const wrapper = mount(
  <Provider store={store}>
    <LoginScreen />
  </Provider>
);

describe('Given the LoginScreen Component', () => {
  test('should renderize & Snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
