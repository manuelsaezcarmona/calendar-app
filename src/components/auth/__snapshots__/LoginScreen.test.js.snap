// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given the LoginScreen Component should renderize & Snapshot 1`] = `
<div
  className="container login-container"
>
  <div
    className="row"
  >
    <div
      className="col-md-6 login-form-1"
    >
      <h3>
        Ingreso
      </h3>
      <form
        onSubmit={[Function]}
      >
        <div
          className="form-group"
        >
          <input
            className="form-control"
            name="logEmail"
            onChange={[Function]}
            placeholder="Correo"
            type="text"
            value="jose@manu.com"
          />
        </div>
        <div
          className="form-group"
        >
          <input
            className="form-control"
            name="logPassw"
            onChange={[Function]}
            placeholder="Contraseña"
            type="password"
            value="123456"
          />
        </div>
        <div
          className="form-group"
        >
          <input
            className="btnSubmit"
            type="submit"
            value="Login"
          />
        </div>
      </form>
    </div>
    <div
      className="col-md-6 login-form-2"
    >
      <h3>
        Registro
      </h3>
      <form
        onSubmit={[Function]}
      >
        <div
          className="form-group"
        >
          <input
            className="form-control"
            name="RegName"
            onChange={[Function]}
            placeholder="Nombre"
            type="text"
            value=""
          />
        </div>
        <div
          className="form-group"
        >
          <input
            className="form-control"
            name="RegEmail"
            onChange={[Function]}
            placeholder="Correo"
            type="email"
            value=""
          />
        </div>
        <div
          className="form-group"
        >
          <input
            className="form-control"
            name="regPassw1"
            onChange={[Function]}
            placeholder="Contraseña"
            type="password"
            value=""
          />
        </div>
        <div
          className="form-group"
        >
          <input
            className="form-control"
            name="regPassw2"
            onChange={[Function]}
            placeholder="Repita la contraseña"
            type="password"
            value=""
          />
        </div>
        <div
          className="form-group"
        >
          <input
            className="btnSubmit"
            type="submit"
            value="Crear cuenta"
          />
        </div>
      </form>
    </div>
  </div>
</div>
`;
